src/modules/previewpane/UnitTests-MonacoPreviewHandler/MonacoPreviewControlTests.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading;
+using System.Windows.Forms;
+using Microsoft.PowerToys.PreviewHandler.Monaco;
+using Microsoft.PowerToys.STATestExtension;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Web.WebView2.WinForms;
+using Moq;
+
+namespace MonacoPreviewHandlerUnitTests
+{
+    [STATestClass]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2201:Do not raise reserved exception types", Justification = "new Exception() is fine in test projects.")]
+    public class MonacoPreviewControlTests
+    {
+        private static readonly int ThreeSecondsInMilliseconds = 3000;
+        private static readonly int SleepTimeInMilliseconds = 200;
+
+        [TestMethod]
+        public void MonacoPreviewControlShouldAddExtendedBrowserControlWhenDoPreviewCalled()
+        {
+            // Arrange
+            using (var monacoPreviewControl = new MonacoPreviewHandlerControl())
+            {
+                // Act
+                monacoPreviewControl.DoPreview<string>("HelperFiles/test.json");
+
+                int beforeTick = Environment.TickCount;
+
+                while (monacoPreviewControl.Controls.Count == 0 && Environment.TickCount < beforeTick + ThreeSecondsInMilliseconds)
```

Comment:
loading label is added synchronously, so no need to wait for it. This loop won't do anything as label is there already after DoPreview call. It will be hard to test "check loading label first and after loading check webview component". let's skip that for now. So you should basically wait for specific type of control (WebView2) here. e.g. if you want to test label, you can do it when file is too big.

Reply:
test actually passed here, but I believe it's just a lucky scenario where everything happened in the correct order/time..

Reply:
As I said webView does never initialize in this test for whatever reason

Reply:
I don't understand? Could you elaborate? This check passed `Assert.IsInstanceOfType(monacoPreviewControl.Controls[0], typeof(WebView2));` in CI. So WebView2 control is added, right?

Reply:
It doesn't pass on my PC. But I'm away from home now for 3 weeks, so I can't give you any further information. If you (or someone else) wants to finish it you can do it.